src/app/profile/page.tsx

```diff
@@ -0,0 +1,143 @@
+  const [authors, { stories, total_pages }] = await Promise.all([
+    authorsPromise,
+    getStories(params),
+  ]);
+  console.log(authors);
+  if (!authors?.roles.includes("AUTHOR")) {
+    console.log("Author: ", id);
+    throw new Error("This person is not an author");
+  }
+
+  return (
+    <>
+      <div className="flex  flex-row">
+        <div className="relative max-h-full  w-full ">
+          <div className="relative h-full bg-white shadow">
+            <div className="flex h-full w-full  justify-between">
+              <div className="fixed flex h-full w-1/6 flex-col bg-[#84B59F] p-6 text-left">
+                <div className="relative ">
+                  <Avatar
+                    imageUrl={"/example.png" ?? undefined}
+                    label="Image"
+                    size="6xl"
+                    className="h-60 w-auto"
+                  />
+                </div>
+                <div className="my-2 text-2xl">
+                  {authors?.firstName} {authors?.lastName}
+                </div>
+                <div className="my-1.5 flex gap-2">
+                  <TopicTag name="BIOLOGY"></TopicTag>
+                  <TopicTag name="CHEMISTRY"></TopicTag>
+                </div>
+                <div className="my-2">
+                  Short Author Bio Here. 1-3 Sentences. Fill text. Fill text.
+                  Fill text. Fill text. Fill text. Fill text. Fill text. Fill
+                  text. Fill text.Fill text. Fill text. Fill text. Fill text.
+                  Fill text. Fill text. Fill text.Fill text.Fill text.Fill text.
+                </div>
+                <div className="mt-2.5 flex">
+                  <FooterIcon type="instagram" />
+                  <FooterIcon type="facebook" />
+                </div>
+              </div>
+              <div className="w-1/6" />
+              <div className="flex w-5/6 p-6 text-center">
+                <div className="h-full">
+                  <ProfileButton searchParams={searchParams} />
+                  <div>
+                    <div className="my-3 text-left">
+                      <HomepageSection
+                        heading={
+                          category ? category + " Stories" : "Recent Stories"
+                        }
+                      />
+                    </div>
+                    <div className="my-3 text-left">
+                      <ArticleList articles={stories} />
+                    </div>
+                    <div className="my-3 text-right">
+                      <Pagination total_pages={total_pages} />
+                    </div>
+                  </div>
+                </div>
+              </div>
+            </div>
+          </div>
+        </div>
+      </div>
+    </>
+  );
+}
+
+async function getStories(params: Record<string, string>) {
+  const searchParams = new URLSearchParams(params);
+  const route = `/stories?${searchParams.toString()}&page_size=30`;
```

Comment:
I keep looking back at /api/stories and for some reason I don't see where it checks the author id in search params.  Am I just missing it?

Reply:
that would mean this is calling the wrong api endpoint, but I'm pretty sure the contributor page should be calling the contributor endpoint?